### PR TITLE
Implements better `toString` for tiles inside Catalyst.

### PR DESCRIPTION
**Pull request by metasim**
Addresses #122

(Review after #123 is merged.)

### CODE REVIEW

File: core/src/main/scala/org/locationtech/rasterframes/tiles/ShowableTile.scala [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    else tile.toArray()
+
+    val cells = if (data.length <= maxCells)
+      data.mkString("[", ",", "]")
+    else {
+      val front = data.take(maxCells/2).mkString("[", ",", "")
+      val back = data.takeRight(maxCells/2).mkString("", ",", "]")
+      front + ",...," + back
+    }
+    s"[${ct.name}, $dims, $cells]"
```
Comment (vpipkt) [inline core/src/main/scala/org/locationtech/rasterframes/tiles/ShowableTile.scala]: Our internal data science team likes this. The "flattened" array representation is fine with us, but I'd expect questions about it from users. It's something we can  address in docs.

**Comment by metasim** [comment]
> In python I still see `RasterRefTile(...` for `spark.read.rastersource(...).show() ` Not sure if that is intended to be totally fixed.

I'm inclined to say "yes", because I want to know the distinction between lazily evaluated tiles, and those that are fully realized.... I suggest letting this sit for a spell and if it causes further consternation we have the code go and read the tile before printing.

File: core/src/main/scala/org/locationtech/rasterframes/tiles/ShowableTile.scala [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+object ShowableTile {
+  private val maxCells = rfConfig.getInt("showable-max-cells")
+  def show(tile: Tile): String = {
+    val ct = tile.cellType
+    val dims = tile.dimensions
+    val data = if (tile.cellType.isFloatingPoint)
+      tile.toArrayDouble()
+    else tile.toArray()
+
+    val cells = if (data.length <= maxCells)
```
[the remark this answers is not in the document]
Comment (metasim) [reply core/src/main/scala/org/locationtech/rasterframes/tiles/ShowableTile.scala]: I don't think it matters unless we first deal with the point below about "take".

File: core/src/main/scala/org/locationtech/rasterframes/tiles/ShowableTile.scala [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      data.mkString("[", ",", "]")
+    else {
+      val front = data.take(maxCells/2).mkString("[", ",", "")
```
[the remark this answers is not in the document]
Comment (metasim) [reply core/src/main/scala/org/locationtech/rasterframes/tiles/ShowableTile.scala]: Let me think on it. Iit might not be as heavy as a lift as I think. Currently we have to fully realize a tile on the JVM (non-Catalyst) side before we can read a single cell... this is due to how tiles are actually stored as byte arrays, and have to be "decoded" to the underlying cell type/size before being able to pull out a single value. But the work on the Python UDT side has refined my thinking on this, and I wonder if we're closer than I might assume.

File: core/src/main/scala/org/locationtech/rasterframes/model/Cells.scala [hunk 4]
```diff
@@ -54,6 +60,7 @@ object Cells {
         Cells(Right(ref.rr))
       case const: ConstantTile =>
         // TODO: Create mechanism whereby constant tiles aren't expanded.
+        // If we don't, the serialization breaks.
```
[the remark this answers is not in the document]
Comment (metasim) [reply core/src/main/scala/org/locationtech/rasterframes/model/Cells.scala]: https://github.com/locationtech/rasterframes/issues/129

File: core/src/main/scala/org/locationtech/rasterframes/tiles/ShowableTile.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (metasim) [reply core/src/main/scala/org/locationtech/rasterframes/tiles/ShowableTile.scala]: No easy solution jumped out at me in 10 minutes, so filing a ticket for further work.

https://github.com/locationtech/rasterframes/issues/130

File: core/src/main/scala/org/locationtech/rasterframes/tiles/ShowableTile.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (metasim) [reply core/src/main/scala/org/locationtech/rasterframes/tiles/ShowableTile.scala]: See https://github.com/locationtech/rasterframes/issues/130

**Pull request merged**